### PR TITLE
WIP: Fix swagger_nickname issue

### PR DESCRIPTION
**Pull request by bcipolli**
https://github.com/caciviclab/disclosure-backend/commit/5ee514a8eb9eb28bc59628c3c80956dbd38af7d0 removed the `swagger_nickname` file, which exposes friendlier names to swagger clients. This led to the error detailed in #149 

In reinstating the old code, I found:
- Our current URL structure leads to strange nicknames for `locality`: `get`, `get_0_1`, `get_0_2`
- If `None` is returned by our function, this causes swagger to raise an error.

The current code:
- Adds tests for `nickname`, to make sure this system is at least minimally verified
- lowercases the default nickname (based on the view name), so that client methods are more meaningful (e.g. `ballot_view_get`).
- never returns `None`

@adborden I'm not sure this is the right fix. Can you review the above and comment? Thanks!

### CODE REVIEW

**Comment by bcipolli** [comment]
OK, I see that if I use a `ViewSet`, I can define different suffixes and avoid the bad default prefixing. I will try this.

File: disclosure/views.py [hunk 1]
```diff
@@ -83,6 +84,7 @@ def location_view(request, locality_id):
         [(key, val.aggregate(tot=Sum(F('amount')))['tot'] or 0)  # 0 for empty
          for key, val in results_by_locality.items()])
 
+    print 'a'
```
Comment (tdooner) [inline disclosure/views.py:87]: Debugging output?

File: ballot/urls/ballot.py [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
 
 admin.autodiscover()
 
-urlpatterns = patterns(
-    '',
-    url(r'^locality/(?P<locality_id>[0-9]+)/ballot/$', views.ballot_view,
-        name='locality_ballot'))
+api = SimpleRouter()
+api.register(r'locality/(?P<locality_id>[0-9]+)', views.BallotViewSet,
+             base_name='ballot')
```
Comment (tdooner) [inline ballot/urls/ballot.py:11]: I think this broke the tests? Anyway, something did -- the failure looks related.

**Comment by tdooner** [comment]
This sounds good to me.

**Comment by adborden** [comment]
On Tue, 2016-01-26 at 16:41 -0800, Ben Cipollini notifications@github.com wrote:

> OK, I see that if I use a `ViewSet`, I can define different suffixes and avoid the bad default prefixing. I will try this.

Yeah, there's not much documentation on this in rest-swagger. It's kind
of a trial and error thing. But I think adding tests for expected
nicknames will make it easier to deal with.

We also might want some extra handling in the `view_name` function to
handle the cases where the nickname would be lame.

## 

Aaron Borden
Human and Hacker

**Comment by bcipolli** [comment]
Closing this PR; what's being done is well beyond a fix for #149

**Pull request closed without merging**